pkg/common/defaulter/trafficmanagerbackend.go

```diff
@@ -0,0 +1,19 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package defaulter
+
+import (
+	"k8s.io/utils/ptr"
+
+	fleetnetv1beta1 "go.goms.io/fleet-networking/api/v1beta1"
+)
+
+// SetDefaultsTrafficManagerBackend sets the default values for TrafficManagerBackend.
+func SetDefaultsTrafficManagerBackend(obj *fleetnetv1beta1.TrafficManagerBackend) {
+	if obj.Spec.Weight == nil {
+		obj.Spec.Weight = ptr.To(int64(1))
+	}
```

Comment:
I wonder if defaulting to 1 makes sense? If there are >1 endpoints, will each receive 0 weight?

Reply:
Also, there is default value in the CEL. I wonder if we still need this.

Reply:
we're following the gateway api and we will take the ceiling, so the weight is 1.

`Also, there is default value in the CEL. I wonder if we still need this.`
In case there is a regression in the crd, we'll set the defaults before handling it to avoid the panic.